components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java

```diff
@@ -46,722 +39,728 @@
+            // direction of the message
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug("Cache-hit for message ID : " + synCtx.getMessageID());
+            }
+            // mark as a response and replace envelope from cache
+            synCtx.setResponse(true);
+            try {
+                if (cachedResponse.isJson()) {
+                    byte[] payload = cachedResponse.getResponsePayload();
+                    OMElement response = JsonUtil.getNewJsonPayload(msgCtx, payload, 0,
+                                                                    payload.length, false, false);
+                    if (msgCtx.getEnvelope().getBody().getFirstElement() != null) {
+                        msgCtx.getEnvelope().getBody().getFirstElement().detach();
+                    }
+                    msgCtx.getEnvelope().getBody().addChild(response);
+
+                } else {
+                    msgCtx.setEnvelope(MessageHelper.cloneSOAPEnvelope(cachedResponse.getResponseEnvelope()));
+                }
+            } catch (AxisFault e) {
+                handleException("Error creating response OM from cache : " + id, synCtx);
+            }
+            if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(getProtocolType())) {
+                msgCtx.setProperty(NhttpConstants.HTTP_SC, Integer.parseInt(cachedResponse.getStatusCode()));
+                msgCtx.setProperty(PassThroughConstants.HTTP_SC_DESC, cachedResponse.getStatusReason());
+            }
+            if (msgCtx.isDoingREST()) {
+
+                msgCtx.removeProperty(NO_ENTITY_BODY);
+                msgCtx.removeProperty(Constants.Configuration.CONTENT_TYPE);
+            }
+            if ((headerProperties = cachedResponse.getHeaderProperties()) != null) {
+
+                msgCtx.setProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS,
+                                   headerProperties);
+                msgCtx.setProperty(Constants.Configuration.MESSAGE_TYPE,
+                                   headerProperties.get(Constants.Configuration.MESSAGE_TYPE));
+            }
+
+            // take specified action on cache hit
+            if (onCacheHitSequence != null) {
+                // if there is an onCacheHit use that for the mediation
+                synLog.traceOrDebug("Delegating message to the onCachingHit "
+                                            + "Anonymous sequence");
+                ContinuationStackManager.addReliantContinuationState(synCtx, 0, getMediatorPosition());
+                if (onCacheHitSequence.mediate(synCtx)) {
+                    ContinuationStackManager.removeReliantContinuationState(synCtx);
+                }
+
+            } else if (onCacheHitRef != null) {
+                if (synLog.isTraceOrDebugEnabled()) {
+                    synLog.traceOrDebug("Delegating message to the onCachingHit " +
```

Comment:
Line wrapping should happen before an operation

Reply:
Fixed in #890